service/advisory_distribution.go

```diff
@@ -0,0 +1,120 @@
+package service
+
+import (
+	"encoding/binary"
+	"fmt"
+	"os"
+	"path/filepath"
+
+	"github.com/rddl-network/distribution-service/config"
+)
+
+func (ds *DistributionService) DistributeToAdvisories() {
+	// gather data
+	currentBlockHeight, err := ds.pmClient.GetBlockHeight()
+	if err != nil {
+		ds.logger.Error("error querying block height:", err.Error())
+		return
+	}
+	lastWrittenBlockHeight, err := ds.ReadLastBlockHeight()
+	if err != nil {
+		ds.logger.Error("error reading block height from file", err.Error())
+		return
+	}
+
+	runDistribution := ds.RunDistribution(currentBlockHeight, lastWrittenBlockHeight)
+	if !runDistribution {
+		ds.logger.Debug("don't run distribution")
+		return
+	}
+
+	err = ds.DistributeToAdvisoriesOnce()
+	if err != nil {
+		return
+	}
+
+	err = ds.WriteLastBlockHeight(currentBlockHeight)
+	if err != nil {
+		ds.logger.Error("error writing to block height file", err.Error())
+		return
+	}
+}
+
+func (ds *DistributionService) DistributeToAdvisoriesOnce() (err error) {
+	cfg := config.GetConfig()
+	distributions := config.GetWeeklyAdvisoryDistribution()
+	for _, distribution := range distributions {
+		address := distribution.Address
+		if cfg.TestnetMode {
+			address = cfg.TestnetAddress
+		}
+		err = ds.sendToAddress(address, distribution.Amount, cfg.Asset)
+		if err != nil {
+			err = fmt.Errorf("sending to address failed: %w", err)
+			return
+		}
+	}
+	return
+}
+
+func (ds *DistributionService) RunDistribution(currentBlockHeight int64, lastWrittenBlockHeight int64) (run bool) {
+	var blocksPerDay int64 = 3600
```

Comment:
does it make sense to hard-code this? If we change the block creation time this will not be correct anymore